### PR TITLE
Correctly handle missing WAGTAILADMIN_BASE_URL setting

### LINKED ISSUE #8582: password reset not working if WAGTAILADMIN_BASE_URL unset
### Issue Summary
After upgrading an old project to wagtail 3.0 where BASE_URL or WAGTAILADMIN_BASE_URL is unset, the password reset functionality of the wagtail admin stops working with the error `KeyError: 'request'` raised from `get_admin_base_url`.

### Steps to Reproduce

1. Have a fresh wagtail install with wagtail 3.0
2. Remove WAGTAILADMIN_BASE_URL setting
3. Create a user account
4. Try the password reset with the email address of the user account
5. You get a 500 with `KeyError: 'request'`

Either `get_admin_base_url` should be fixed to work in emails when WAGTAILADMIN_BASE_URL is unset or it should be made clearer in the release notes, that you need a WAGTAILADMIN_BASE_URL setting.

* I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: yes

### Technical details

* Python version: 3.10
* Django version: 4.0.4
* Wagtail version: 3.0
* Browser version: Chrome 101

### PR DESCRIPTION
**Pull request by gasman**
Fixes #8582 as per https://github.com/wagtail/wagtail/issues/8582#issuecomment-1154244542 - remove logic that attempts to construct the base URL from the request (because the request generally isn't available).

### CODE REVIEW

File: wagtail/admin/checks.py [hunk 1]
```diff
@@ -164,3 +164,22 @@ def check_panels_in_model(cls, context="model"):
         errors.append(error)
 
     return errors
+
+
+@register("wagtail_admin_base_url")
```
Comment (lb-) [inline wagtail/admin/checks.py:169]: should we create a future task to prefix all the other checks' names?

File: wagtail/admin/checks.py [hunk 2]
```diff
@@ -164,3 +164,22 @@ def check_panels_in_model(cls, context="model"):
...
+    if getattr(settings, "WAGTAILADMIN_BASE_URL", None) is None:
+        errors.append(
+            Warning(
+                "The WAGTAILADMIN_BASE_URL setting is not defined",
+                hint="This should be the base URL used to access the Wagtail admin site. "
+                "Without this, URLs in notification emails will not display correctly",
+                id="wagtailadmin.W003",
+            )
+        )
+
```
Comment (lb-) [inline wagtail/admin/checks.py:184]: will we need to add something to upgrade considerations?

**Review by lb- — COMMENTED** [review]
Have not run locally but changes look good and I agree that it's better for the config go be explicit instead of attempting to pull it from context. 

I'll try to run it locally tomorrow if no one else looks at it. 

A reminder to check the notifications (e.g. workflow submitted) - I think I recall some odd bugs for this setting when notifications are sent. Hence why the context workaround existed.

File: wagtail/admin/checks.py (hunk 2 above)
Comment (gasman) [reply wagtail/admin/checks.py:184]: I think https://docs.wagtail.org/en/stable/releases/3.0.html#base-url-setting-renamed-to-wagtailadmin-base-url probably covers everything we need to say already - this warning just makes it more visible.

File: wagtail/admin/checks.py (hunk 1 above)
Comment (gasman) [reply wagtail/admin/checks.py:169]: I'm not intending this to set a precedent of all our checks starting with `wagtail_`, if that's what you mean - it just seemed logical to give it the same name as the setting it was checking for. (So really it should be `wagtailadmin_base_url` - I kept tripping up on the underscores while writing this, and I guess this one got left in...)

**Review by lb- — APPROVED** [review]
Thanks Matt.

Really appreciate your eyes on this, I had missed this nuance (despite my testing notifications) on the original PR.

The try/catch improvement is also super helpful.

**Pull request merged**